### PR TITLE
fix: telegraf.service startup script on sles15 during instalation

### PR DESCRIPTION
**Pull request by sanekkurt**
https://github.com/influxdata/telegraf/issues/4507 solved
Due to the absence of the /etc/SuSE-release file, telegraph.service was not started in the new versions of sles 15. Therefore, the logic for running the service in linux sles was added to the logic for checking the $NAME variable. Now the script can work both in old versions of linux sles and in new versions.

### CODE REVIEW

**Comment by sanekkurt** [comment]
!signed-cla

File: scripts/rpm/post-remove.sh [hunk 1]
```diff
@@ -44,5 +44,16 @@ elif [[ -f /etc/os-release ]]; then
...
+        if [[ "$1" = "0" ]]; then
+            rm -f /etc/default/telegraf
+
+            if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+                disable_systemd /usr/lib/systemd/system/telegraf.service
+            else
+                # Assuming sysv
+                disable_chkconfig
+            fi
+        fi
```
Comment (reimda) [inline scripts/rpm/post-remove.sh:57]: This is copied from line 20.  Would you put it in a function and call the function in both places?  Maybe name the function disable_systemd_or_chkconfig

File: scripts/rpm/post-install.sh [hunk 2]
```diff
@@ -84,5 +84,18 @@ elif [[ -f /etc/os-release ]]; then
...
+        else
+            # Assuming SysVinit
+            install_init
+            # Run update-rc.d or fallback to chkconfig if not available
+            if which update-rc.d &>/dev/null; then
+                install_update_rcd
+            else
+                install_chkconfig
+            fi
+        fi
```
Comment (reimda) [inline scripts/rpm/post-install.sh:99]: This is copied from line 58.  Would you put it in a function and call it from both places?  Maybe name the function install_systemd_or_chkconfig

**Review by reimda — CHANGES_REQUESTED** [review]

File: scripts/rpm/post-remove.sh (hunk 1 above)
Comment (sanekkurt) [reply scripts/rpm/post-remove.sh]: Yes, of course. Added a function in a new commit

File: scripts/rpm/post-install.sh (hunk 2 above)
Comment (sanekkurt) [reply scripts/rpm/post-install.sh]: Yes, of course. Added a function in a new commit

File: scripts/rpm/post-install.sh [hunk 3]
```diff
@@ -24,6 +24,22 @@ function install_chkconfig {
     chkconfig --add telegraf
 }
 
+function install_systemd_or_chkonfig {
```
Comment (reimda) [inline scripts/rpm/post-install.sh:27]: Thanks!  But it looks like there's a typo here and also in the disable function 
```suggestion
function install_systemd_or_chkconfig {
```

File: scripts/rpm/post-remove.sh [hunk 4]
```diff
@@ -15,7 +15,7 @@ function disable_chkconfig {
     rm -f /etc/init.d/telegraf
 }
 
-if [[ -f /etc/redhat-release ]] || [[ -f /etc/SuSE-release ]]; then
+function disable_systemd_or_chkonfig {
```
Comment (reimda) [inline scripts/rpm/post-remove.sh:18]: :)
```suggestion
function disable_systemd_or_chkconfig {
```

**Review by reimda — CHANGES_REQUESTED** [review]

File: scripts/rpm/post-install.sh (hunk 3 above)
Comment (sanekkurt) [reply scripts/rpm/post-install.sh]: Sorry I didn't notice the mistake )

**Comment by sspaink** [comment]
@sanekkurt sorry for not getting back to you for a while, are you still interested in working on this PR? Could you re-base it and we can give it a final review.

**Comment by reimda** [comment]
The bot didn't notice new commits from two days ago so I'm reopening

**Comment by powersj** [comment]
Hi,

I'm going to close this PR as a fix for #4507 has landed. That fix makes the install scripts OS-agnostic to avoid needing to add specific distro names.

Thanks!

**Pull request closed without merging**